cumulusci/tasks/bulkdata/step.py

```diff
@@ -75,7 +75,7 @@ def download_file(uri, bulk_api):
         resp = requests.get(uri, headers=bulk_api.headers(), stream=True)
         resp.raise_for_status()
         f = os.fdopen(handle, "wb")
-        for chunk in resp.iter_content(chunk_size=None):
```

Comment:
@davisagli - cumulusci/tasks/tests/test_step.py::TestDownloadFile::test_download_file() was hanging after updates to the `responses` package. @codexmarcel and I looked at [the source code](https://github.com/psf/requests/blob/b0e025ade7ed30ed53ab61f542779af7e024932e/requests/models.py#L780)for this method and suspect this might be a bug? Having a chunk size of none doesn't seem to make sense.

Reply:
The docs for requests say: "A value of None will function differently depending on the value of stream. stream=True will read data as it arrives in whatever size the chunks are received." So the use of None is intentional here to make sure that we read in all data that has been received so far, not just one byte at a time (chunk_size defaults to 1). I wouldn't like to make this change without checking its impact on performance.

This may need further investigation to figure out if there's a bug in responses that we should report.

Reply:
I'll take a look this morning to see if I can repro in a simpler script.

Reply:
Here is some (slightly) simplified code to repro the the failure. When debugging I'm seeing that [this while loop never exits](https://github.com/urllib3/urllib3/blob/fac7db19a1e8e15e5b302e2203ce35cd0485bc88/src/urllib3/response.py#L726).
```
import os
import pathlib
import requests
import responses
import tempfile


def download_file(uri, handle, chunk_size):
    """Download the Bulk API result file for a single batch,
    and remove it when the context manager exits."""
    resp = requests.get(uri, headers={}, stream=True)
    resp.raise_for_status()
    f = os.fdopen(handle, "wb")

    for chunk in resp.iter_content(chunk_size=chunk_size):
        f.write(chunk)

    f.close()
    return f


@responses.activate
def test_download_file__chunk_size_one():
    url = "https://example.com"
    responses.add(method="GET", url=url, body=b"TEST\xe2\x80\x94")

    (handle, path) = tempfile.mkstemp(text=False)
    f = download_file(url, handle, chunk_size=1)
    pathlib.Path(path).unlink()


@responses.activate
def test_download_file__chunk_size_none():
    url = "https://example.com"
    responses.add(method="GET", url=url, body=b"TEST\xe2\x80\x94")

    (handle, path) = tempfile.mkstemp(text=False)
    f = download_file(url, handle, chunk_size=None)
    pathlib.Path(path).unlink()
```

Reply:
@Br4nd0R I wonder if that is fixed by this change in responses that hasn't been released yet: https://github.com/getsentry/responses/pull/422

Reply:
Pinning to 0.14.0 for now.